src/api/Flooq.Api/Controllers/DataFlowController.cs

```diff
@@ -0,0 +1,120 @@
+#nullable disable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Flooq.Api.Domain;
+using Flooq.Api.Model;
+
+namespace Flooq.Api.Controllers
+{
+    [Route("api/dataflow")]
+    [ApiController]
+    public class DataFlowController : ControllerBase
+    {
+        private readonly FlooqContext _context;
+
+        public DataFlowController(FlooqContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/DataFlow
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<DataFlow>>> GetDataFlows()
+        {
+            return await _context.DataFlows.ToListAsync();
+        }
+
+        // GET: api/DataFlow/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<DataFlow>> GetDataFlow(Guid id)
+        {
+            var dataFlow = await _context.DataFlows.FindAsync(id);
+
+            if (dataFlow == null)
+            {
+                return NotFound();
+            }
+
+            return dataFlow;
+        }
+
+        // PUT: api/DataFlow/5
+        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutDataFlow(Guid id, DataFlow dataFlow)
+        {
+            if (id != dataFlow.Id)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(dataFlow).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!DataFlowExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return NoContent();
+        }
+
+        // POST: api/DataFlow
+        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        [HttpPost("{useDemoFlow}")]
+        public async Task<ActionResult<DataFlow>> PostDataFlow(bool useDemoFlow, DataFlow dataFlow)
+        {
+          if (useDemoFlow)
+          {
+            dataFlow.Name = "Demo Flow";
+            string sample_data = "";
+            using (StreamReader reader = new StreamReader("sampleDataFlow.json"))
```

Comment:
Is the demo flow intended to remain or is this just for our own testing purposes? In the latter case, please remove it before the final merge.

Reply:
I discussed this with Felix. The demo data flow is now implemented using [Data Seeding](https://docs.microsoft.com/en-us/ef/core/modeling/data-seeding) in FlooqContext.cs.